test/topology_custom/test_table_desc_read_barrier.py

```diff
@@ -55,7 +55,15 @@ async def test_table_desc_read_barrier(request, manager: ManagerClient) -> None:
     async with inject_error(manager.api, servers[0].ip_addr, 'group0_state_machine::delay_apply'):
         logger.info("Altering table")
         await cql.run_async("alter table ks.t add s1 int", host=hosts[1])
-        # TODO: trigger the read barrier once available
-        desc_schema = [await cql.run_async("DESC SCHEMA", host=h) for h in hosts]
 
+        # verify that there is a schema difference
+        # (raft apply delayed, so the first server doesn't have the new schema yet)
+        desc_schema = [await cql.run_async("DESC SCHEMA", host=h) for h in hosts]
         assert desc_schema[0] != desc_schema[1]
+
+        # waits for the schema fully updated on all nodes (after the delay ends)
```

Comment:
@kbr-scylla @gleb-cloudius is this comment true - does a read barrier wait for **all** nodes? Isn't it just a quorum?

Reply:
A read barrier only waits for the node it is executed on to see the latest state.

Reply:
In fact it should only wait for the schema to be updated on the node where the read barrier is issued (i.e. on the first node) - I assume at least

Reply:
After read barrier successfully finishes on node X, it is guaranteed that node X has applied all group 0 commands that were committed before this read barrier started. (And maybe more if concurrently more commands were being committed.)

Read barrier implementation does require contacting a quorum (the impl RPCs the leader and then the leader RPCs a quorum -- roughly speaking) but it doesn't say anything about whether this quorum will apply the commands, only node X.

Reply:
fixed the comment